File: src/cpp/FindGreatestSumOfSubArray.cpp
class Solution {
public:
    int FindGreatestSumOfSubArray(vector<int> array) {
    	int n = array.size();
        if (n == 0) return 0;

        int maxSum = array[0];
        int curSum = 0;
        for (int i = 0; i < n; ++i) {
            curSum = (curSum + array[i] > array[i]) ? curSum + array[i] : array[i];
            maxSum = (maxSum > curSum) ? maxSum : curSum;
        }
        return maxSum;
    }
};

File: src/java/FindGreatestSumOfSubArray.java
public class Solution {
    public int FindGreatestSumOfSubArray(int[] array) {
        if (array == null || array.length == 0) {
            return 0;
        }
        int currSum = 0;
        int greatestSum = Integer.MIN_VALUE;
        for (int i = 0; i < array.length; i++) {
            if (currSum <= 0) {
                currSum = array[i];		// 记录当前最大值
            } else {
                currSum += array[i];	// 当array[i]为正数时，加上之前的最大值并更新最大值。
            }

            if (currSum > greatestSum) {
                greatestSum = currSum;
            }
        }
        return greatestSum;
    }
}
